handler.go

```diff
@@ -39,14 +39,15 @@ type oauthUserInfo struct {
 	Email     string
 }
 
-type userModel struct {
+// UserModel holds the user information in the claim
+type UserModel struct {
```

Comment:
I had to export this as I was running into this issue (https://github.com/golang/go/issues/21357), which as far as I understand is an issue with older versions of Go where Unmarshal was able to set unexported fields, but this has been fixed in newer versions of Go which must explain why I'm seeing this now. This project uses https://github.com/deis/docker-go-dev/releases/tag/v1.5.0 in the Dockerfile, which uses Go 1.9, I'm using 1.11 locally. We might want to update this.

Reply:
from the issue you point, it seems that it should be enough instantiating the inner struct:

```
a := &userClaims{}
a.userModel = &userModel{}
```

It seems a bit weird that this struct is the only one exported, if the error was introduced with the changes in this PR I would try the above. If that error appears without changes then it's fine to leave this exported.

Reply:
I might be misunderstanding you, but note that I'm not instantiating the struct directly, the error comes with the JSON Unmarshall which doesn't have access to unexported fields. What you suggested works in the package if I were to directly create the struct, but won't work in another package or using reflect (see the issue I linked to).

Actually, the struct JSON Unmarshall tries to fill is:

```
type userClaims struct {
	*UserModel
	jwt.StandardClaims
}
```

so jwt.StandardClaims is exported, and now UserModel is too. Actually, I can avoid exporting the UserModel struct by just exporting the field inside `userClaims`:

```
type userClaims struct {
	UserModel *userModel
	jwt.StandardClaims
}
```

I'll update it to this as this is cleaner and doesn't leave an exported field that we don't really need. Thanks!

Reply:
Also I can confirm that using Go 1.9.1, the same version that the container is built with, and reverting these changes back to using the unexported field, the tests work. But, again on my local machine with Go 1.11, they don't. So this confirms that this is caused by a bug that was present in 1.9.1 and fixed later, and that the correct way to do this is to use an exported field.

Reply:
FYI, here is the error you get with 1.11 with the unexported field:

```
json: cannot set embedded pointer to unexported struct: main.userModel
```

Reply:
> Actually, I can avoid exporting the UserModel struct by just exporting the field inside `userClaims`:
>
>```
>type userClaims struct {
>	UserModel *userModel
>	jwt.StandardClaims
>}
>```
>I'll update it to this as this is cleaner and doesn't leave an exported field that we don't really >need. Thanks!

Actually, on further investigation, this won't work because now the `UserModel` struct doesn't get inlined in the JSON, and is now nested in a `"UserModel": {...}` key. I also tried adding the `json:",inline"` tag but that didn't work. We'll need to keep this as-is.

Reply:
I see, thanks for the research, it was not clear for me where it was failing.